controller/lib/core/blower_pid.cpp

```diff
@@ -65,6 +72,32 @@ void blower_pid_init() {
   myPID.SetMode(AUTOMATIC);
 }
 
+static void update_volume(SensorReadings *readings) {
+  // TODO: This calculation should be much more sophisticated.  Some possible
+  // improvements.
+  //
+  //  - Periodically re-zero the volume (e.g. what happens if the tubes are
+  //    disconnected from the patient?)
+  //
+  //  - Measure time with better than millisecond granularity.
+  Time now = Hal.now();
+  Duration delta = now - last_flow_measurement_time;
+  bool first_time = isnan(last_flow_ml_per_min);
+  if (!first_time && delta <= VOLUME_INTEGRAL_INTERVAL) {
```

Comment:
Maybe merge this condition with the one below, to make what happens first time and non-first time into independently readable blocks.

Reply:
Thanks for that; I wrote it this way to avoid repeating the two lines
```
    last_flow_ml_per_min = readings->flow_ml_per_min;
    last_flow_measurement_time = now;
```
but I was not really comfortable with how I did it.